contracts/contracts/gatekeepers/SignUpTokenGatekeeper.sol

```diff
@@ -2,16 +2,17 @@
 pragma solidity ^0.8.20;
 
 import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 
 import { SignUpGatekeeper } from "./SignUpGatekeeper.sol";
-import { SignUpToken } from "../SignUpToken.sol";
 
 /// @title SignUpTokenGatekeeper
 /// @notice This contract allows to gatekeep MACI signups
 /// by requiring new voters to own a certain ERC721 token
 contract SignUpTokenGatekeeper is SignUpGatekeeper, Ownable(msg.sender) {
   /// @notice the reference to the SignUpToken contract
-  SignUpToken public token;
+  ERC721 public immutable token;
```

Comment:
Do we have any plans for supporting other token types?

Reply:
mm not sure, this one was meant to work with ERC721 specifically, but eventually we should be getting rid of gatkeepers within MACI and using Excubiae's https://github.com/privacy-scaling-explorations/zk-kit.solidity/tree/main/packages/excubiae